addons/popochiu/engine/objects/graphic_interface/components/dialog_text/portrait_text_panel.gd

```diff
@@ -0,0 +1,20 @@
+extends PanelContainer
+
+@onready var portrait_text: RichTextLabel = %PortraitText
+@onready var left_avatar: TextureRect = $HBoxContainer/LeftAvatar
+@onready var right_avatar: TextureRect = $HBoxContainer/RightAvatar
+
+
+func _ready() -> void:
+	# Connect to child signals
+	portrait_text.text_show_started.connect(show)
+	portrait_text.text_show_finished.connect(hide)
+	
+	# Connect to singletons signals
+	C.character_spoke.connect(_update_avatar)
+	
+	hide()
+
+
+func _update_avatar(chr: PopochiuCharacter, _msg := '') -> void:
+	left_avatar.texture = chr.get_avatar_for_emotion(chr.emotion)
```

Comment:
Instead of writing a TODO, I decided to implement a default behavior that will render the avatar of the talking character depending on its position on screen. If the character is positioned on the left side, the left avatar will be used, otherwise, the right avatar will be used.